libminifi/include/utils/StringUtils.h

```diff
@@ -347,6 +227,139 @@ class StringUtils {
     return join(std::basic_string<TChar>(separator), container);
   };
 
+  /**
+   * Hexdecodes the hexencoded string in data, ignoring every character that is not [0-9a-fA-F]
+   * @param data the output buffer where the hexdecoded bytes will be written. Must be at least length / 2 bytes long.
+   * @param data_length pointer to the length of data the data buffer. It will be filled with the length of the decoded bytes.
+   * @param hex the hexencoded string
+   * @param hex_length the length of hex
+   * @return true on success
+   */
+  inline static bool from_hex(uint8_t* data, size_t* data_length, const char* hex, size_t hex_length) {
+    if (*data_length < hex_length / 2) {
+      return false;
+    }
+    uint8_t n1;
+    bool found_first_nibble = false;
+    *data_length = 0;
+    for (size_t i = 0; i < hex_length; i++) {
+      const uint8_t byte = static_cast<uint8_t>(hex[i]);
+      if (byte > 127) {
+        continue;
+      }
+      uint8_t n = hex_lut[byte];
+      if (n != SKIP) {
+        if (found_first_nibble) {
+          data[(*data_length)++] = n1 << 4 | n;
+          found_first_nibble = false;
+        } else {
+          n1 = n;
+          found_first_nibble = true;
+        }
+      }
+    }
+    if (found_first_nibble) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Hexdecodes a string
+   * @param hex the hexencoded string
+   * @param hex_length the length of hex
+   * @return the vector containing the hexdecoded bytes
+   */
+  inline static std::vector<uint8_t> from_hex(const char* hex, size_t hex_length) {
+    std::vector<uint8_t> decoded(hex_length / 2);
```

Comment:
Sure?
If hex_length is odd, this buffer becomes too small by the division. 
I know that the scenario is odd, too.

Reply:
If the hex_length is odd (and does not contain any filler characters), then the encoding is invalid (there will be an incomplete byte). In this case the above from_hex function will not assemble and write this byte to the buffer and returns with false instead, so there is no buffer overflow.
There is also a test case specifically for this.

Reply:
Okay, thanks!